### PR TITLE
feat: encode_pretty

### PR DESCRIPTION
**Pull request by enghitalo**
```v
struct Foo2 {
	ux8  u8
	ux16 u16
	ux32 u32
	ux64 u64
	sx8  i8
	sx16 i16
	sx32 int
	sx64 i64
	a    bool
	b    string
	c    Foo[string]
	d    StringAlias
}

fn test_pretty() {
	foo := Foo2{1, 2, 3, 4, -1, -2, -3, -4, true, 'abc', Foo[string]{
		name: 'aa'
		data: 'bb'
	}, 'aliens'}
	assert json.encode_pretty(foo) == '{
  "ux8": 1,
  "ux16": 2,
  "ux32": 3,
  "ux64": 4,
  "sx8": -1,
  "sx16": -2,
  "sx32": -3,
  "sx64": -4,
  "a": true,
  "b": "abc",
  "c": {
    "name": "aa",
    "data": "bb"
  },
  "d": "aliens"
}'
}
```

### CODE REVIEW

**Comment by Delta456** [comment]
Can you add more examples with different data types? I think it would be better.

**Comment by enghitalo** [comment]
> Can you add more examples with different data types? I think it would be better.

Yes. Just a minute

**Comment by enghitalo** [comment]
@Delta456. Thank you for the sugestion

**Comment by enghitalo** [comment]
The failed test pass in my local (Ubuntu)
```v
./v test vlib/x/json2                                                                                                                                                                                                                                                                               
---- Testing... ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 OK    [1/9]   307.124 ms vlib/x/json2/integer_primitives_test.v
 OK    [2/9]   316.074 ms vlib/x/json2/encoder_test.v
 OK    [3/9]   349.499 ms vlib/x/json2/decoder_test.v
 OK    [4/9]   353.789 ms vlib/x/json2/json_module_compatibility_test/json_decode_with_optional_arg_test.v
 OK    [5/9]   354.620 ms vlib/x/json2/json_module_compatibility_test/json_decode_with_encode_arg_test.v
 OK    [6/9]   367.207 ms vlib/x/json2/json2_test.v
 OK    [7/9]   371.921 ms vlib/x/json2/json_module_compatibility_test/json_test.v
 OK    [8/9]   375.219 ms vlib/x/json2/any_test.v
 OK    [9/9]   380.847 ms vlib/x/json2/scanner_test.v
 ```

File: vlib/x/json2/json_module_compatibility_test/json_todo_test.vv [hunk 1]
```diff
@@ -434,3 +423,71 @@ fn test_encoding_struct_with_pointers() {
 	}
 	assert json.encode(value) == '{"association":{"price":{}},"price":{}}'
 }
+
+struct Foo2 {
```
Comment (spytheman) [inline vlib/x/json2/json_module_compatibility_test/json_todo_test.vv:427]: This struct is not used at all in this test.

**Review by spytheman — APPROVED** [review]

**Pull request merged**